src/daily_output.ts

```diff
@@ -0,0 +1,474 @@
+import { readFileSync, writeFileSync } from 'fs';
+import { GROUP_EFFORT_REWARDS, ONE_THOUSAND_MB, OUTPUT_NAME, OUTPUT_TEMPLATE_NAME } from './constants';
+import { GQLEdgeInterface } from './gql_types';
+import { OutputData, StakedPSTHolders, WalletsStats, WalletStatEntry } from './inferno_types';
+
+/**
+ * A class responsible of parsing the GQL and CommunityOracle data into the OutputData file
+ */
+export class DailyOutput {
+	private readonly previousData = this.read();
+	private data = this.read();
+	private latestBlock = 0;
+	private latestTimestamp = 0;
+	private bundlesTips: { [txId: string]: number } = {};
+	private bundlesToVerify: { [txId: string]: { walletAddress: string; fileSize: number }[] } = {};
+
+	/**
+	 * Takes the data from the previously generated data, fallbacking to the base template if not present
+	 * @throws if the validation of the output file fails
+	 * @returns {OutputData}
+	 */
+	public read(): OutputData {
+		const data = (() => {
+			try {
+				return this.readOutputFile();
+			} catch (err) {
+				console.info(`The output file hasn't yet been created. Using the file template`);
+				return this.readTemplate();
+			}
+		})();
+		if (!this.validateDataStructure(data)) {
+			throw new Error(`The output JSON has a wrong structure`);
+		}
+		return data;
+	}
+
+	/**
+	 * @param {StakedPSTHolders} stakedPSTHolders key/value of address/tokens above 200 ARDRIVE locked for at least 21600 blocks (~30 days)
+	 */
+	feedPSTHolders(stakedPSTHolders: StakedPSTHolders): void {
+		this.data.PSTHolders = stakedPSTHolders;
+	}
+
+	/**
+	 * @param {GQLEdgeInterface[]} queryResult the edges of ArFSTransactions only - 50 block before the latest
+	 */
+	feedGQLData(queryResult: GQLEdgeInterface[]): Promise<void> {
```

Comment:
For code reading consistency:

```suggestion
	public feedPSTHolders(stakedPSTHolders: StakedPSTHolders): void {
		this.data.PSTHolders = stakedPSTHolders;
	}

	/**
	 * @param {GQLEdgeInterface[]} queryResult the edges of ArFSTransactions only - 50 block before the latest
	 */
	public feedGQLData(queryResult: GQLEdgeInterface[]): Promise<void> {
```

It will help to understand what gets used in the above scope